backend/src/common/util/execution_status.go

```diff
@@ -14,13 +14,30 @@
 
 package util
 
+type ExecutionPhase string
+
+const (
+	ExecutionUnknown   ExecutionPhase = ""
+	ExecutionPending   ExecutionPhase = "Pending" // pending some set-up - rarely used
+	ExecutionRunning   ExecutionPhase = "Running" // any node has started; pods might not be running yet, the workflow maybe suspended too
+	ExecutionSucceeded ExecutionPhase = "Succeeded"
+	ExecutionFailed    ExecutionPhase = "Failed" // it maybe that the the workflow was terminated
+	ExecutionError     ExecutionPhase = "Error"
+)
```

Comment:
How do decide this is the right set of state? Can you point me to the current Argo workflow state? Thanks!

Reply:
https://pkg.go.dev/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1#WorkflowPhase

Reply:
I can put this link as comment